examples/download.rs

```diff
@@ -54,6 +64,32 @@ impl Cli {
             .context("the firmware file is too big")?;
         file.seek(io::SeekFrom::Start(0))?;
 
+        let mut bar: Option<indicatif::ProgressBar> = None;
```

Comment:
It might be easier to call once DfuLibusb::open and see if you get an error, and then enter the loop. This would avoid the logic about setting the progress bar :sweat_smile:

Reply:
it just ends up with 2 calls to open; but indeed this way around did turn out a bit nicer i think ; nice suggestion